### PR TITLE
Add check for overlapping points in neurites

### PR DESCRIPTION
**Pull request by adrien-berchet**

### CODE REVIEW

**Review by arnaudon — APPROVED** [review]

File: neurom/check/morphtree.py [hunk 1]
```diff
@@ -204,6 +206,60 @@ def is_back_tracking(neurite):
...
+                    np.concatenate(
+                        [np.ones((len(sec.points) - 1, 1)) * sec.id, sec.points[1:]],
+                        axis=1,
+                    )
+                    for sec in neurite.iter_sections()
+                ],
+            ),
+        ],
+    )
+    tree = KDTree(section_pts[:, [1, 2, 3]])
```
Comment (eleftherioszisis) [inline neurom/check/morphtree.py:234]: Fun fact:
```
section_pts[:, [1, 2, 3]]
```
performs a copy
```
section_pts[:, 1:4]
```
doesn't.

File: neurom/check/morphology_checks.py [hunk 2]
```diff
@@ -367,11 +367,11 @@ def has_no_back_tracking(morph):
     return CheckResult(len(bad_ids) == 0, bad_ids)
 
 
-def has_no_duplicated_point(morph, tolerance=None):
-    """Check if the morphology has duplicated points."""
+def has_no_overlapping_point(morph, tolerance=None):
+    """Check if the morphology has overlapping points."""
```
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply neurom/check/morphology_checks.py:371]: Sure

File: neurom/check/morphtree.py [hunk 3]
```diff
@@ -204,6 +206,60 @@ def is_back_tracking(neurite):
...
+                [
+                    np.concatenate(
+                        [np.ones((len(sec.points) - 1, 1)) * sec.id, sec.points[1:]],
+                        axis=1,
+                    )
+                    for sec in neurite.iter_sections()
+                ],
+            ),
+        ],
+    )
```
[the remark this answers is not in the document]
Comment (adrien-berchet) [reply neurom/check/morphtree.py:233]: Ok :innocent:

File: neurom/check/morphtree.py (hunk 1 above)
Comment (adrien-berchet) [reply neurom/check/morphtree.py:234]: :scream:

**Comment by adrien-berchet** [comment]
Thanks!

**Pull request merged**